File: Car Pooling.js
var carPooling = function(trips, capacity) {
 trips.sort((a, b) => a[1] - b[1])
   
        trip = Array(1000).fill(0)
    
    for(let i = 0; i < trips.length; i++) {
        for(let j = trips[i][1]; j < trips[i][2]; j++) {
            trip[j] += trips[i][0]
            if(trip[j] > capacity) return false;
        }
    }
    return true;
};

File: 1184-car-pooling/car-pooling.py
class Solution(object):
    def carPooling(self, trips, capacity):
        passenger = [0] * 1001
    
        for trip in trips:
            pass_count, start, end = trip
            passenger[start] += pass_count
            passenger[end] -= pass_count
        current_passengers = 0
        for count in passenger:
            current_passengers += count
            if current_passengers > capacity:
                return False
        return True








        